libs/common/src/lib/retrieval-agent/agent-dashboard/workflow/basic-elements/node-form/node-form.component.html

```diff
@@ -0,0 +1,159 @@
+<div *ngIf="!isFormReady" class="loading-indicator">
+  Loading...
+</div>
+
+<app-configuration-form
+  *ngIf="isFormReady"
+  [form]="form"
+  (cancel)="cancel.emit()"
+  (triggerSubmit)="submit()">
+  <ng-container [formGroup]="configForm">
+    <ng-container *ngIf="schemaKeys.length">
```

Comment:
Coding style: to avoid having `ng-container` everywhere, it is cleaner to use this syntax:
```
@if (schemaKeys.length) {
    <some>stuff</some>
}
```